File: Source/Aladdin.ASN1.ISO/ASN1/ISO/OCSP/OCSPResponse.cs
using System;
using System.Runtime.Serialization;

// OCSPResponse ::= SEQUENCE {
//    responseStatus              OCSPResponseStatus,
//    responseBytes  [0] EXPLICIT ResponseBytes OPTIONAL
// }

namespace Aladdin.ASN1.ISO.OCSP
{
	[Serializable]
	public class OCSPResponse : Sequence
	{
		// информация о структуре
		private static readonly ObjectInfo[] info = new ObjectInfo[] { 

			new ObjectInfo(new ObjectCreator<Enumerated<OCSPResponseStatus>>().Factory(), Cast.N                 ), 
			new ObjectInfo(new ObjectCreator<ResponseBytes			       >().Factory(), Cast.EO, Tag.Context(0)) 
		}; 
		// конструктор при сериализации
        protected OCSPResponse(SerializationInfo info, StreamingContext context) : base(info, context) {}

		// конструктор при раскодировании
		public OCSPResponse(IEncodable encodable) : base(encodable, info) {} 

		// конструктор при закодировании
		public OCSPResponse(Enumerated<OCSPResponseStatus> responseStatus, ResponseBytes ResponseBytes) : 
			base(info, responseStatus, ResponseBytes) {} 

		public Enumerated<OCSPResponseStatus>	ResponseStatus	{ get { return (Enumerated<OCSPResponseStatus>)this[0]; } } 
		public ResponseBytes					ResponseBytes	{ get { return (ResponseBytes                 )this[1]; } }
	}
}

File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/ocsp/OCSPResponse.java
package aladdin.asn1.iso.ocsp;
import aladdin.asn1.*;
import java.io.*;

// OCSPResponse ::= SEQUENCE {
//    responseStatus              OCSPResponseStatus,
//    responseBytes  [0] EXPLICIT ResponseBytes OPTIONAL
// }

public class OCSPResponse extends Sequence<IEncodable>
{
    private static final long serialVersionUID = -4592579215281690593L;

    // информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(Enumerated   .class).factory(), Cast.N                 ), 
		new ObjectInfo(new ObjectCreator(ResponseBytes.class).factory(), Cast.EO, Tag.context(0)) 
	}; 
	// конструктор при раскодировании
	public OCSPResponse(IEncodable encodable) throws IOException { super(encodable, info); }
    
	// конструктор при закодировании
	public OCSPResponse(Enumerated responseStatus, ResponseBytes responseBytes) 
	{ 
		super(info, responseStatus, responseBytes); 
	}
	public final Enumerated    responseStatus() { return (Enumerated   )get(0); } 
	public final ResponseBytes responseBytes () { return (ResponseBytes)get(1); }
}
